internal/pkgs/wifimanager/controller.go

```diff
@@ -5,100 +5,299 @@ package wifimanager
+				if err != nil {
+					return output, fmt.Errorf("failed to get Strength: %s", err)
+				}
+				wpaFlags_variant, err := ap.GetProperty("org.freedesktop.NetworkManager.AccessPoint.WpaFlags")
+				wpaFlags_variant.Store(&wpaFlags)
+				if err != nil {
+					return output, fmt.Errorf("failed to get WPA flags: %s", err)
+				}
+				rsnFlags_variant, err := ap.GetProperty("org.freedesktop.NetworkManager.AccessPoint.RsnFlags")
+				rsnFlags_variant.Store(&rsnFlags)
+				if err != nil {
+					return output, fmt.Errorf("failed to get RSN flags: %s", err)
+				}
+				// Convert variables to string and append it into output
+				output.Ssid = append(output.Ssid, string(ssid))
+				output.Signal = append(output.Signal, strength)
+				output.Security = append(output.Security, wpaFlags+rsnFlags)
+			}
+		}
+	}
+	return output, nil
 }
 
 func (c *WifiController) Connect(ctx context.Context, SSID string, Password string) (string, error) {
+	var devicePaths []dbus.ObjectPath
+	var response string
 
 	// Input validation
 	if ctx == nil {
 		return "", fmt.Errorf("context cannot be nil")
 	}
 
-	nmcliRunCmd := "/run/current-system/sw/bin/nmcli"
-	nmcliRunCmd += " device wifi connect "
-	nmcliRunCmd += SSID
-	nmcliRunCmd += " password \""
-	nmcliRunCmd += Password
-	nmcliRunCmd += "\""
+	conn, err := dbus.ConnectSystemBus()
+	if err != nil {
+		return "", fmt.Errorf("failed to connect to system bus: %s", err)
+	}
+	defer conn.Close()
 
-	cmd := exec.Command("/bin/sh", "-c", nmcliRunCmd)
-	response, err := cmd.CombinedOutput()
+	obj := conn.Object("org.freedesktop.NetworkManager", "/org/freedesktop/NetworkManager")
+	err = obj.Call("org.freedesktop.NetworkManager.GetDevices", 0).Store(&devicePaths)
+	if err != nil {
+		return "", fmt.Errorf("failed to get devices: %s", err)
+	}
+
+	// Iterate over devices to find Wi-Fi devices
+	for _, devicePath := range devicePaths {
+		var deviceType uint32
+
+		device := conn.Object("org.freedesktop.NetworkManager", devicePath)
+		deviceTypeVriant, err := device.GetProperty("org.freedesktop.NetworkManager.Device.DeviceType")
+		if err != nil {
+			return "", fmt.Errorf("failed to get device type: %s", err)
+		}
+		err = deviceTypeVriant.Store(&deviceType)
+		if err != nil {
+			return "", fmt.Errorf("failed to convert device type: %s", err)
+		}
+
+		// Check if the device is a Wi-Fi device (type 2)
+		if deviceType == 2 {
+			var connectionPath dbus.ObjectPath
+			var activeConnectionPath dbus.ObjectPath
+
+			settings := map[string]map[string]dbus.Variant{
+				"connection": {
+					"id":          dbus.MakeVariant(SSID),
+					"type":        dbus.MakeVariant("802-11-wireless"),
+					"autoconnect": dbus.MakeVariant(true),
+				},
+				"802-11-wireless": {
+					"ssid":     dbus.MakeVariant([]byte(SSID)),
+					"mode":     dbus.MakeVariant("infrastructure"),
+					"security": dbus.MakeVariant("802-11-wireless-security"),
+				},
+				"802-11-wireless-security": {
+					"key-mgmt": dbus.MakeVariant("wpa-psk"),
```

Comment:
Should this be dynamic to not limit us to a specific key-mgmt type? E.g., "sae",  "wpa-eap", ... (https://networkmanager.dev/docs/api/latest/settings-802-11-wireless-security.html)

Reply:
I have tried to implement some branching from NetworkManager. I couldn't check enterprise protocol but it seems tat wpa personals works. I can separate it within another PR if you prefer that.